src/main/java/com/artipie/SliceFromConfig.java

```diff
@@ -273,4 +269,41 @@ static Slice build(final Settings settings, final Authentication auth,
             .<Slice>map(limit -> new ContentLengthRestriction(slice, limit))
             .orElse(slice);
     }
+
+    /**
+     * Creates Docker proxy repository slice from configuration.
+     *
+     * @param cfg Repository configuration.
+     * @return Docker proxy slice.
+     */
+    private static Slice dockerProxy(final RepoConfig cfg) {
+        final YamlMapping settings = cfg.settings()
+            .orElseThrow(() -> new IllegalStateException("Repo settings not found"));
+        final String host = settings.string("host");
+        if (host == null) {
+            throw new IllegalStateException("`host` is not specified in settings");
+        }
+        final Credentials credentials;
+        final String username = settings.string("username");
+        final String password = settings.string("password");
+        if (username == null && password == null) {
+            credentials = Credentials.ANONYMOUS;
+        } else {
+            if (username == null) {
+                throw new IllegalStateException("`username` is not specified in settings");
+            }
+            if (password == null) {
+                throw new IllegalStateException("`password` is not specified in settings");
+            }
```

Comment:
@olegmoz also I'd suggest to extend exception messages with repository type, otherwise it may not be clear what setting to check